sys/net/routing/aodvv2/aodv.c

```diff
@@ -63,22 +64,20 @@ static mutex_t rerr_mutex;
 
 struct netaddr na_mcast;
 kernel_pid_t aodvv2_if_id;
-ipv6_addr_t aodvv2_prefix;
+ng_ipv6_addr_t aodvv2_prefix;
 int aodvv2_prefix_len;
 
-void aodv_init(void)
+void aodv_init(kernel_pid_t interface)
 {
     AODV_DEBUG("%s()\n", __func__);
 
-    /* init this thread's IPC msg queue (TODO: do I need to do this?) */
+    /* init this thread's IPC msg queue */
     msg_t msgq[RCV_MSG_Q_SIZE];
     msg_init_queue(msgq, sizeof msgq);
 
-    /* TODO: set & handle prefix and prefix_len properly (consider AODVV2_RIOT_PREFIXLEN!) */
-    aodvv2_prefix_len = 0;
-    /* TODO: set if_id properly (as param of aodv_init) */
-    aodvv2_if_id = 0;
-    net_if_set_src_address_mode(aodvv2_if_id, NET_IF_TRANS_ADDR_M_SHORT);
+    aodvv2_prefix_len = sizeof(ng_ipv6_addr_t);
```

Comment:
+1 for `address_type_size`
> Maybe we should also change the variable names/documentation in the fib?

I will change the name of `prefix_size` to `prefix_address_type_size` [1] this afternoon. 
(sorry for the misleading name and documentation)

[1] https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/ng_fib.h#L80